source/Dictionary.h

```diff
@@ -20,25 +20,106 @@ this program. If not, see <https://www.gnu.org/licenses/>.
 #include <utility>
 #include <vector>
 
+// Code from:
+// https://gist.github.com/filsinger/1255697/1972eb676b47116838edaacf923e60b9173199c2
+namespace hash_fnv
+{
+	using def_type = uint32_t;
+
+	template <typename S> struct fnv_internal;
+	template <typename S> struct fnv1;
+	template <typename S> struct fnv1a;
+
+	template <> struct fnv_internal<def_type>
+	{
+		constexpr static def_type default_offset_basis = 0x811C9DC5;
+		constexpr static def_type prime = 0x01000193;
+	};
+
+	template <> struct fnv1<def_type> : public fnv_internal<def_type>
+	{
+		constexpr static inline def_type hash(char const *const aString, const def_type val = default_offset_basis)
+		{
+			return (aString[0] == '\0') ? val : hash(&aString[1], (val * prime) ^ def_type(aString[0]));
+		}
+	};
+
+	template <> struct fnv1a<def_type> : public fnv_internal<def_type>
+	{
+		constexpr static inline def_type hash(char const *const aString, const def_type val = default_offset_basis)
+		{
+			return (aString[0] == '\0') ? val : hash(&aString[1], (val ^ def_type(aString[0])) * prime);
+		}
+	};
```

Comment:
You can simplify this massively:

1. Why are the functions inside a struct?
2. `fnv1` is unused, since you're using `fnv1a` only
3. To avoid having the constants in global namespace, I think it's fine if you inline them, since they're only used once and are fairly opaque anyways.

Reply:
Yes, you're right: I only waited to know if the algorithm I choose is ok...
At that time I take the first one I saw :grin:, which reveals a discrete luck as this one still quite fast and really simple...